src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -179,6 +176,45 @@ public WebElement waitForElement(String id) {
         return waitForElement(PageUtil.buildBy(configuration, id));
     }
 
+    public WebElement waitForElementByAccessibilityIdentifier(AccessibilityIdentifier id) {
+        try {
+            waitForCondition(ExpectedConditions.not(ExpectedConditions.invisibilityOf(driver.findElementByAccessibilityId(id.getId()))));
+        } catch (Exception e) {
+            Logger.info("WaitForElement: Eat exception thrown waiting for condition");
+        }
+
+        int size = driver.findElementsByAccessibilityId(id.getId()).size();
```

Comment:
I'd change this to a list instead of the size, then if you find the item(s) later in the while look you can just save them to this variable and break out of the loop.